src/js/components/DataTable/ExpanderCell.js

```diff
@@ -35,11 +46,68 @@ const ExpanderControl = ({ context, expanded, onToggle, pad, ...rest }) => {
   );
 
   if (onToggle) {
+    const expandText =
+      messages?.expand ||
+      format({
+        id: 'dataTable.expand',
+        messages: {
+          'dataTable.expand': 'expand',
+        },
+      });
+
+    const collapseText =
+      messages?.collapse ||
+      format({
+        id: 'dataTable.collapse',
+        messages: {
+          'dataTable.collapse': 'collapse',
+        },
+      });
+
+    const expandAllText =
+      messages?.expandAll ||
+      format({
+        id: 'dataTable.expandAll',
+        messages: {
+          'dataTable.expandAll': 'expand all',
+        },
+      });
+
+    const collapseAllText =
+      messages?.collapseAll ||
+      format({
+        id: 'dataTable.collapseAll',
+        messages: {
+          'dataTable.collapseAll': 'collapse all',
+        },
+      });
+
+    let a11yTitle;
+
+    if (context === 'header') {
+      a11yTitle = expanded ? collapseAllText : expandAllText;
+    } else {
+      a11yTitle = expanded ? collapseText : expandText;
+      if (expandAriaLabel) {
+        let ariaLabel;
+        if (typeof expandAriaLabel === 'function') {
+          if (context === 'groupHeader') {
+            ariaLabel = expandAriaLabel(row.datum);
+          } else {
+            ariaLabel = expandAriaLabel(row);
+          }
+        } else {
+          ariaLabel = expandAriaLabel;
+        }
+        a11yTitle = `${a11yTitle} ${ariaLabel}`;
```

Comment:
This looks to me like it's trying to compose a phrase by concatenating several parts. This is something we need to avoid when internationalizing. Text should be combined via substitution parameters for the overall phrase so localizers can order things as best makes sense for the language.